lib/plugins/aws/deploy/index.js

```diff
@@ -80,6 +80,13 @@ class AwsDeploy {
               .getStage()} ${style.aside(`(${this.serverless.getProvider('aws').getRegion()})`)}`
           );
           log.info(); // Ensure gap between verbose logging
+
+          // This is used to ensure that for `deploy` command, the `accountId` will be resolved and available
+          // for `generatePayload` telemetry logic
+          this.serverless
+            .getProvider('aws')
+            .getAccountId()
+            .then((accountId) => (this.serverless.getProvider('aws').accountId = accountId));
```

Comment:
- `this.serverless .getProvider('aws')`  is accessible at `this.provider` why not rely on that?
- `getAccountId()` resolves with `accountId`, why not simply take it from there?
- Let's silence eventual rejection (otherwise it'll throw as unhandled in case of e.g. missing credentials)

Reply:
1. Good point
2. I'm not sure what do you mean by that - could you explain?
3. of course :facepalm:

Reply:
> I'm not sure what do you mean by that - could you explain?

Sorry, I misread the code, ignore this one